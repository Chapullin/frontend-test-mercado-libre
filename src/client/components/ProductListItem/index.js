import React from 'react';
import { Link } from 'react-router-dom';
import shipping from '../../../assets/ic_shipping.png';

export default class ProductListItem extends React.Component {
  constructor(props) {
    super(props);
  }

  render() {
    return (
      <Link to={`/items/${this.props.item.id}`}>
        <section className="results-item">
          <section className="results-item__thumbnail">
            <img className="img" src={this.props.item.thumbnail} alt="" />
          </section>
          <section className="results-item__info">
            <h1 className="results-item__price">
                $
              {this.props.item.price}
              {this.props.item.free_shipping ? <img src={shipping} alt=""/> : ''}
            </h1>

            <p className="results-item__location">
              {this.props.item.address.city_name}
            </p>
            <h2 className="results-item__title">{this.props.item.title}</h2>
          </section>
        </section>
      </Link>
    );
  }
}
