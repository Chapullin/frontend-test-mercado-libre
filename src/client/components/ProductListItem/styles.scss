.results-item {
  display: inline-flex;
  padding: $spacing-16;
  position: relative;
  width: 100%;
  color: $meli-dark;

  &__thumbnail {
    padding-right: $spacing-16;
    width: 180px;
    height: 180px;

    .img {
      height: 100%;
      border: 1px solid $meli-light-gray;
    }
  }

  &__price {
    font-size: 24px;
    margin-top: 0;
    margin-bottom: $spacing-32;
  }

  &__location {
    color: $meli-gray;
    font-size: 12px;
    position: absolute;
    top: 0;
    right: 16px;
  }

  &__title {
    font-size: 18px;
  }
}