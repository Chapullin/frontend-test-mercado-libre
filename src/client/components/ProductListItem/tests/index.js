import React from "react";
import renderer from "react-test-renderer";
import ProductListItem from "../../ProductListItem";
import { BrowserRouter as Router, Route, Link, Switch } from "react-router-dom";

describe("ProductListItem", () => {
  it("should match snapshot - base", () => {
    const item = {
      id: 'MLA93423034',
      thumbnail: 'image.jpg',
      address: {
        city_name: 'Rosario',
      }
    }
    const component = renderer
      .create(
        <Router>
          <ProductListItem item={item} />
        </Router>
      )
      .toJSON();
    expect(component).toMatchSnapshot();
  });
});
