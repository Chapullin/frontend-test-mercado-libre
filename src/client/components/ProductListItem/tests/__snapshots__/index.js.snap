// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductListItem should match snapshot - base 1`] = `
<a
  href="/items/MLA93423034"
  onClick={[Function]}
>
  <section
    className="results-item"
  >
    <section
      className="results-item__thumbnail"
    >
      <img
        alt=""
        className="img"
        src="image.jpg"
      />
    </section>
    <section
      className="results-item__info"
    >
      <h1
        className="results-item__price"
      >
        $
        
      </h1>
      <p
        className="results-item__location"
      >
        Rosario
      </p>
      <h2
        className="results-item__title"
      />
    </section>
  </section>
</a>
`;
